File: C++/calculator.cpp
#include <iostream>
#include <string>
using namespace std;

void calculator();

int main()
{
    // Call the calculator function.
    calculator();
    return 0;
}

// Calculates the result of a simple arithmetic expression entered by the user.
void calculator()
{
    // Get the first number.
    cout << "Welcome to the calculator program." << endl;
    cout << "Enter your first number: ";
    float number1;
    cin >> number1;

    // Get the second number.
    cout << "Enter your second number: ";
    float number2;
    cin >> number2;

    // Get the operator.
    cout << "Enter your operator: ";
    string op;
    cin >> op;

    // Perform the calculation.
    if (op == "+")
    {
        cout << "The result is: " << (number1 + number2) << endl;
    }
    else if (op == "-")
    {
        cout << "The result is: " << (number1 - number2) << endl;
    }
    else if (op == "*")
    {
        cout << "The result is: " << (number1 * number2) << endl;
    }
    else if (op == "/")
    {
        cout << "The result is: " << (number1 / number2) << endl;
    }
    else
    {
        cout << "Invalid input. Please try again." << endl;
    }
}

File: Javascript/calculator.js
function calculator() {
	console.log("Welcome to the calculator program.");
	let number1 = prompt("Enter your first number: ");
	let number2 = prompt("Enter your second number: ");
	let operator = prompt("Enter your operator: ");

	if (operator === "+") {
		console.log(
			"The result is: " + (parseFloat(number1) + parseFloat(number2))
		);
	} else if (operator === "-") {
		console.log(
			"The result is: " + (parseFloat(number1) - parseFloat(number2))
		);
	} else if (operator === "*") {
		console.log("The result is: " + parseFloat(number1) * parseFloat(number2));
	} else if (operator === "/") {
		console.log("The result is: " + parseFloat(number1) / parseFloat(number2));
	} else {
		console.log("Invalid input. Please try again.");
	}
}

calculator();
